cluster/images/etcd/migrate/versions_test.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"testing"
+)
+
+func TestSerializeEtcdVersionPair(t *testing.T) {
+	cases := []struct {
+		versionTxt string
+		version    *EtcdVersionPair
+		match      bool
+	}{
+		{"3.1.2/etcd3", &EtcdVersionPair{&EtcdVersion{3, 1, 2}, storageEtcd3}, true},
```

Comment:
Shall we add some more random cases, like completely out of nowhere number of etcd version and etcdstorage version, like 23.1.6 and etcd6? As well as nonserialized etcd version pair? I know the code handles these scenario, but just for completeness of the test.

Reply:
Sure, I've expanded the test coverage.